.github/workflows/deploy-containers.yml

```diff
@@ -3,6 +3,8 @@ on:
   workflow_dispatch:
   push:
     branches: ['main', 'ghcr_base_image']
+  pull_request:
+    branches: ['main']
```

Comment:
I think we discussed this already at some point. I can see that this way, the base image is updated in the GHCR so that the github build action can pull the latest image from GHCR. But I don't think we should push to the GHCR before the PR is checked. I strongly recommend building the base image within the github action with the committed code and only deploying reviewed base images. I think the GHCR should always be synchronized to main. If you update the GHCR with a PR, this is not the case anymore.

Reply:
See #99